src/main/groovy/edu/oregonstate/mist/api/Resource.groovy

```diff
@@ -145,9 +147,8 @@ abstract class Resource {
      * @return the url
      */
     protected String getPaginationUrl(Map params, String resourceEndpoint) {
-        String path = endpointUri.path + resourceEndpoint
-
-        URIBuilder uriBuilder = new URIBuilder(endpointUri).setPath(path)
+        URI baseUri = UriBuilder.fromUri(endpointUri).path(resourceEndpoint).build()
```

Comment:
UriBuilder preserves the original path of the uri so I'm using that to build a uri, then basing the URIBuilder object off the path of baseUri.